src/client/search/AddSearchTerm.js

```diff
@@ -22,12 +22,15 @@ let AddSearchTerm = ({ dispatch }) => {
         onBlur={() => {
           $('#searchTermContainer').slideUp('fast');
         }}
+
         onKeyDown={(e) => {
           if (e.keyCode === 13) {
             dispatch(addSearchTerm(nextSearchTermId++, e.target.value));
+            dispatch(invalidateFeedResults());
```

Comment:
It depends on who we consider the "owner" of the search terms to be - but I'm happy to go with option number 2, as you say it's more decoupled that way.